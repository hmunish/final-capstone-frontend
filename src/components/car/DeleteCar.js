import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { deleteCar, getCars } from "../../redux/cars/carsSlice";

const DeleteCar = () => {
  const { cars, isLoading, isUpdated } = useSelector((state) => state.cars);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getCars());
  }, [dispatch]);

  if (isUpdated) {
    dispatch(getCars());
  }

  const handleDeleteCar = (carId) => {
    dispatch(deleteCar({ carId }));
  };

  if (isLoading) {
    return (
      <div>
        <h1>Loading...</h1>
      </div>
    );
  }

  if (cars) {
    return (
      <section className="delete-cars-list">
        <div className="green-overlay" />
        <div className="add-car-form">
          {cars.map((car) => (
            <div className="delete-car-item" key={car.id}>
              <p>{car.name}</p>
              <button className="delete-btn" onClick={() => handleDeleteCar(car.id)} type="button">
                Delete
                {car.name}
              </button>
            </div>
          ))}
        </div>
      </section>
    );
  }

  return <p>Error loading cars</p>;
};

export default DeleteCar;
