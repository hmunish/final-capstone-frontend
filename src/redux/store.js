import { configureStore } from '@reduxjs/toolkit';
import loginReducer from './login/loginSlice';
import signupReducer from './signup/signupSlice';

const store = configureStore({
  reducer: {
    login: loginReducer,
    signup: signupReducer,
  },
});

export default store;
