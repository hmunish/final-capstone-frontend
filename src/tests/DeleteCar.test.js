import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import DeleteCar from '../components/car/DeleteCar';
import store from '../redux/store';

test('Car Details components Match', () => {
  const tree = renderer.create(
    <BrowserRouter>
      <Provider store={store}>
        <DeleteCar />
      </Provider>
    </BrowserRouter>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
