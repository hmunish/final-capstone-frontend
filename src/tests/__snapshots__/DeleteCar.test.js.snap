// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Car Details components Match 1`] = `
<section
  className="delete-cars-list"
>
  <div
    className="green-overlay"
  />
  <div
    className="cars-to-delete"
  >
    <h1
      className="delete-title"
    >
      Delete Cars
    </h1>
  </div>
</section>
`;
